src/hooks/useLinearPrimaryTicks.ts

```diff
@@ -0,0 +1,98 @@
+import type { ScaleContinuousNumeric } from 'd3-scale';
+import { MutableRefObject, useEffect, useMemo, useState } from 'react';
+
+import { textDimentions } from '../utils';
+
+type Directions = 'horizontal' | 'vertical';
+
+export interface PrimaryLinearTicks {
+  label: string;
+  position: number;
+}
+
+interface Options {
+  tickFormat?: (d: number) => string;
+  scientificNotation?: boolean;
```

Comment:
Please remove this option, the tickFormat should handle this if the user needs it